core/vm/memory.go

```diff
@@ -81,6 +82,31 @@ func (m *Memory) GetCopy(offset, size int64) (cpy []byte) {
 	return
 }
 
+const (
+	memoryPadLimit = 1024 * 1024
+)
+
+// GetMemoryCopyPadded returns offset + size as a new slice.
+// It zero-pads the slice if it extends beyond memory bounds.
+func GetMemoryCopyPadded(m *Memory, offset, size int64) ([]byte, error) {
```

Comment:
I don't think this tracer-helper method belongs in `core/vm`. Why not put it in `eth/tracers`?  `Memory` exposes `Len()`, so there's afaict nothing private that you need access to in order to replicate this functionality somewhere else.